src/matrixeventprocessor.ts

```diff
@@ -92,6 +98,25 @@ export class MatrixEventProcessor {
         return body;
     }
 
+    public ReplaceDiscordEmoji(content: string, guild: Discord.Guild): string {
+        console.log("Gonna replace")
+        let results = DISCORD_EMOJI_REGEX.exec(content);
+        while (results !== null) {
+            const emojiName = results[1];
+            const emojiNameWithColons = results[0];
+
+            // Check if this emoji exists in the guild
+            if(guild.emojis[emojiName] !== null) {
+                // Replace :a: with <:a:123ID123>
+                const emojiID = guild.emojis.find((emoji) => { return emoji.name === emojiName }).id;
```

Comment:
Oooh, isn't the above ``guild.emojis[emojiName]`` already an emoji object you can peel the ``id`` from. Why are you doing two lookups?